src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignContigsAndCallBreakpointsSpark.java

```diff
@@ -0,0 +1,121 @@
+     * over-partitioned since each worker needs to localize the BWA reference index and read it into
+     * memory once per partition.
+     */
+    public static final int NUM_ASSEMBLIES_PER_PARTITION = 200;
+
+    @Argument(doc = "file for assembled breakpoint output", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    private String output;
+
+    @Argument(doc = "Input file of assembled contigs", shortName = "inputFile",
+            fullName = "inputFile", optional = false)
+    private String input;
+
+    private static final Logger log = LogManager.getLogger(AlignContigsAndCallBreakpointsSpark.class);
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+
+        final JavaRDD<String> inputAssemblies = ctx.textFile(input).cache();
+
+        final long numInputPartitions = inputAssemblies.count();
+
+        final JavaPairRDD<String, String> contigCollectionByBreakpointId =
+                inputAssemblies
+                        .flatMapToPair(AlignContigsAndCallBreakpointsSpark::splitAssemblyLine)
+                        .coalesce((int) Math.ceil((double) numInputPartitions / (double) NUM_ASSEMBLIES_PER_PARTITION));
+
+        final JavaPairRDD<String, RunSGAViaProcessBuilderOnSpark.ContigsCollection> breakpointIdsToContigsCollection =
+                contigCollectionByBreakpointId.mapValues(AlignContigsAndCallBreakpointsSpark::parsePackedFasta);
+
+        final JavaPairRDD<String, ContigAligner.AssembledBreakpoint> assembledBreakpoints = breakpointIdsToContigsCollection.mapPartitionsToPair(iter -> () -> {
+            try {
+                final String ref = referenceArguments.getReferenceFileName();
+                final File localRef = BucketUtils.isHadoopUrl(ref) ? BwaSparkEngine.localizeReferenceAndBwaIndexFiles(ref) : new File(ref);
+                try (ContigAligner contigAligner = new ContigAligner(localRef)) {
+                    final List<Tuple2<String, ContigAligner.AssembledBreakpoint>> results = new ArrayList<>();
+                    iter.forEachRemaining(cc -> {
+                        final List<ContigAligner.AssembledBreakpoint> breakpointAssemblies = contigAligner.alignContigs(cc._2);
+                        breakpointAssemblies.forEach(b -> results.add(new Tuple2<>(cc._1, b)));
+                    });
+                    return results.iterator();
+                }
+            } catch (IOException e) {
+                throw new GATKException("Cannot run BWA-MEM", e);
+            }
+
+        });
+
+        assembledBreakpoints.saveAsTextFile(output);
+
+    }
+
+    /**
+     * To ease file format difficulties
+     * @param v
+     * @return
+     */
+    @VisibleForTesting
+    protected static RunSGAViaProcessBuilderOnSpark.ContigsCollection parsePackedFasta(final String v) {
```

Comment:
changed name to `packedFastaLine` and moved to `RunSGAProcessBuilderOnSpark .ContigsCollection`